client/src/components/menubar/index.tsx

```diff
@@ -266,249 +257,218 @@ class MenuBar extends React.PureComponent<MenuBarProps, State> {
     });
   };
 
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  handleSubset = () => {
-    const { dispatch } = this.props;
-
+  const handleSubset = () => {
     track(EVENTS.EXPLORER_SUBSET_BUTTON_CLICKED);
 
     dispatch(actions.subsetAction());
   };
 
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  handleSubsetReset = () => {
-    const { dispatch } = this.props;
-
+  const handleSubsetReset = () => {
     track(EVENTS.EXPLORER_RESET_SUBSET_BUTTON_CLICKED);
 
     dispatch(actions.resetSubsetAction());
   };
 
-  render() {
-    const {
-      dispatch,
-      disableDiffexp,
-      clipPercentileMin,
-      clipPercentileMax,
-      graphInteractionMode,
-      showCentroidLabels,
-      categoricalSelection,
-      colorAccessor,
-      subsetPossible,
-      subsetResetPossible,
-      screenCap,
-      imageUnderlay,
-    } = this.props;
-    const { pendingClipPercentiles } = this.state;
-
-    const isColoredByCategorical =
-      !!categoricalSelection?.[colorAccessor || ""];
-
-    const isTest = getFeatureFlag(FEATURES.TEST);
-    const isDownload = getFeatureFlag(FEATURES.DOWNLOAD);
-
-    // constants used to create selection tool button
-    const [selectionTooltip, selectionButtonIcon] = [
-      "select",
-      "polygon-filter",
-    ];
-
-    return (
-      <div
+  const isColoredByCategorical = !!categoricalSelection?.[colorAccessor || ""];
+
+  const isTest = getFeatureFlag(FEATURES.TEST);
+  const isDownload = getFeatureFlag(FEATURES.DOWNLOAD);
+
+  // constants used to create selection tool button
+  const [selectionTooltip, selectionButtonIcon] = ["select", "polygon-filter"];
+
+  // if you add or remove buttons from the menubar, you must also update the size array constant
+  const rightMenuBarComponents = [
```

Comment:
I put the menu components in this array as they'll be split between the row and column views.